### DISCUSSION TITLE
16bit png images for classification?

### DISCUSSION THREAD

**Discussion by andife** [ludwig-ai/ludwig#2377] [Q&A]
Hi,
I wonder if anyone has patched https://github.com/ludwig-ai/ludwig/blob/master/ludwig/utils/image_utils.py to work with 16bit png images? (At the moment 16bit png images do not work)

I know there are some problems with the support in Pytorch (uint16 is not supported 
https://github.com/pytorch/pytorch/issues/58734),
but torchvision managed the handling with that
https://github.com/pytorch/vision/issues/2218, https://github.com/pytorch/vision/pull/4657

Are there any plans for adding that? Would it be enough to adjust the loading of the images, or would you then also have to adjust the model etc. accordingly. Another question with image augmentation does not arise, because that would have to be done externally by ludwig anyway?

Thank you

**Comment by ShreyaR** [comment]
Hi @andife, thanks for raising this issue. After reading through the linked PyTorch issues, I believe it should be feasible to support 16-bit PNG images. As a stopgap solution, we can use the private API `_decode_16bits_png()` (referenced in https://github.com/pytorch/vision/issues/4731).
